spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/couchbase/CouchbaseProperties.java

```diff
@@ -41,8 +48,28 @@
 
 	private final Env env = new Env();
 
+	private int bootstrapHttpDirectPort = DefaultCoreEnvironment.BOOTSTRAP_HTTP_DIRECT_PORT;
```

Comment:
What are those two properties? Why are they necessary?

The current keys are structure and those two "boostrap"  related keys aren't. They aren't documented either.

Reply:
When CouchbaseMock is initialized on a random available port, these 2 ports are also changed from defaults to available ones. For a couchbase-client to be able to connect to CouchbaseMock we need to set these properties to the ports on which CouchbaseMock is actually listening.

Reply:
So those properties must be set on the builder when `CouchbaseMock` is active but they don't need  to be exposed for the general use case? Again, that's calling for making those changes internally rather than exposing them.

Reply:
you mean `DefaultCouchbaseEnvironment.Builder` ? yes, that would be much better. thank you